tsunami/tracking.py

```diff
@@ -32,6 +33,35 @@ def suspend():
         state.suspended = False
 
 
+def mutable_signal_receiver(func):
+    """
+    Decorator for signals to allow them to be skipped by setting the attr MUTE_SIGNALS_ATTR on an instance.
+    """
+    @functools.wraps(func)
+    def wrapper(sender, instance, signal, **kwargs):
+        mute_signals = getattr(instance, app_settings.MUTE_SIGNALS_ATTR, False)
+        if mute_signals is True:
+            pass # Skip all signals
+        elif isinstance(mute_signals, list) and signal in mute_signals:
+            pass # Skip user requested signal
+        else:
+            return func(sender=sender, instance=instance, signal=signal, **kwargs) # Allow signal receiver
+    return wrapper
+        
+
+@contextlib.contextmanager
+def mute_signals_for(instance, sigs):
+    """
+    Context manager to skip signals for @instance (django model).
+
+    @sigs can be True to skip all signals or list of specified signals like [post_delete, post_save]
+    """
```

Comment:
```suggestion
    """
    Context manager to mute any decorated signals which run within.
    
    Within this context handler, signals decorated with mutable_signal_receiver (see above)
    for the instance (django model class) given will be muted.
    
    Only signals which have been decorated will be prevented.
    
    The second argument, sigs, allows you to choose which decorated signals to mute.
    If True, all decorated signals will be muted. Otherwise, provide a list of signals to mute,
    like [post_delete, post_save]
    """
```